### PR TITLE
Use java fx coordinates for hit test

### PR DESCRIPTION
**Pull request by 1-alex98**

### CODE REVIEW

**Comment by 1-alex98** [comment]
Main change is that mouse and window coordinates are taken from stage and mouse coordinates from ROBOT

**Comment by 1-alex98** [comment]
Fixes #5

File: src/main/java/ch/micheljung/fxwindow/DecorationWindowProcedure.java [hunk 1]
```diff
@@ -123,87 +195,12 @@ public LRESULT callback(HWND hwnd, int uMsg, WPARAM wparam, LPARAM lParam) {
...
-    }
-
-    double scaledTopInset = topInset / Screen.getPrimary().getOutputScaleY();
-    double scaledMouseX = mouse.x / Screen.getPrimary().getOutputScaleX();
-    double scaledMouseY = mouse.y / Screen.getPrimary().getOutputScaleY();
+  /**
+   * Determines which area is used for resize and which area is used for dragging.
+   */
+  private LRESULT hitTest() {
+    Point2D mousePosition = ROBOT.getMousePosition();
```
Comment (1-alex98) [inline src/main/java/ch/micheljung/fxwindow/DecorationWindowProcedure.java]: use javafx robot for mouse

File: src/main/java/ch/micheljung/fxwindow/DecorationWindowProcedure.java [hunk 2]
```diff
@@ -123,87 +195,12 @@ public LRESULT callback(HWND hwnd, int uMsg, WPARAM wparam, LPARAM lParam) {
...
 
-    // Can't use bounds.contains() because it deals with doubles not integers, which causes rounding issues
-    return scaledMouseX >= bounds.getMinX() && scaledMouseX <= bounds.getMaxX()
-      && scaledMouseY >= bounds.getMinY() + scaledTopInset && scaledMouseY <= bounds.getMaxY() + scaledTopInset;
+    return new LRESULT(hitTest(new Rect(windowController.getStage()), new Point(mousePosition), windowController, features.isAllowTopResize()).windowsValue);
```
Comment (1-alex98) [inline src/main/java/ch/micheljung/fxwindow/DecorationWindowProcedure.java:202]: use javafx stage for window

**Comment by 1-alex98** [comment]
When testing it it's smooth and works

File: src/main/java/ch/micheljung/fxwindow/DecorationWindowProcedure.java [hunk 3]
```diff
@@ -75,6 +74,79 @@
     references.add(this);
   }
 
+  static HitTestResult hitTest(Rect window, Point mouse, WindowController controller, boolean allowTopResize) {
```
Comment (micheljung) [inline src/main/java/ch/micheljung/fxwindow/DecorationWindowProcedure.java]:
```suggestion
  private static HitTestResult hitTest(Rect window, Point mouse, WindowController controller, boolean allowTopResize) {
```

File: src/main/java/ch/micheljung/fxwindow/DecorationWindowProcedure.java [hunk 4]
```diff
@@ -59,6 +57,7 @@
   private static final User32Ex user32Ex = User32Ex.INSTANCE;
 
   public static final LRESULT LRESULT_ZERO = new LRESULT(0);
+  public static final Robot ROBOT = new Robot();
```
Comment (micheljung) [inline src/main/java/ch/micheljung/fxwindow/DecorationWindowProcedure.java]:
```suggestion
  private static final Robot robot = new Robot();
```
Only constant-like fields are UPPERCASE

File: src/main/java/ch/micheljung/fxwindow/DecorationWindowProcedure.java [hunk 5]
```diff
@@ -2,26 +2,24 @@
...
-import static com.sun.jna.platform.win32.WinDef.HWND;
-import static com.sun.jna.platform.win32.WinDef.LPARAM;
-import static com.sun.jna.platform.win32.WinDef.LRESULT;
-import static com.sun.jna.platform.win32.WinDef.RECT;
-import static com.sun.jna.platform.win32.WinDef.WPARAM;
+import static com.sun.jna.platform.win32.WinDef.*;
 import static com.sun.jna.platform.win32.WinUser.GWL_WNDPROC;
 import static com.sun.jna.platform.win32.WinUser.WM_DESTROY;
 
+@Log4j2
```
Comment (micheljung) [inline src/main/java/ch/micheljung/fxwindow/DecorationWindowProcedure.java]: Why? I don't see it being used.
```suggestion
```

Remove import, too.

**Review by micheljung — CHANGES_REQUESTED** [review]

**Comment by 1-alex98** [comment]
ok yeah sure gonna fix those... I was foremost interested in an evaluation of my approach

File: src/main/java/ch/micheljung/fxwindow/DecorationWindowProcedure.java [hunk 6]
```diff
@@ -59,6 +55,7 @@
   private static final User32Ex user32Ex = User32Ex.INSTANCE;
 
   public static final LRESULT LRESULT_ZERO = new LRESULT(0);
+  public static final Robot robot = new Robot();
```
Comment (micheljung) [inline src/main/java/ch/micheljung/fxwindow/DecorationWindowProcedure.java]:
```suggestion
  private static final User32Ex user32Ex = User32Ex.INSTANCE;
  private static final Robot robot = new Robot();

  public static final LRESULT LRESULT_ZERO = new LRESULT(0);
```

**Review by micheljung — CHANGES_REQUESTED** [review]

**Review by micheljung — APPROVED** [review]

**Comment by 1-alex98** [comment]
You have to make a new tag I guess @micheljung

**Comment by micheljung** [comment]
@1-alex98 I finally managed to release 0.8.1 to maven central

**Pull request merged**